src/components/OptionRow.js

```diff
@@ -76,6 +77,12 @@ const defaultProps = {
 };
 
 const OptionRow = (props) => {
+    const [isDisabled, setIsDisabled] = React.useState(props.isDisabled);
```

Comment:
Sorry for not calling it out in the proposal phase, but please refer to our [JavaScript style guide](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#stateless-components-vs-pure-components-vs-class-based-components-vs-render-props---when-to-use-what) and the rest of the code in this repo. We have not yet adopted the use of React hooks in this codebase, so this needs to be converted into a class component for now.

```js
class OptionRow extends Component {
    constructor(props) {
        super(props);      
        this.state = {
            isDisabled: this.props.isDisabled,
        };
    }
    
    componentDidUpdate(prevProps) {
        if (this.props.isDisabled === prevProps.isDisabled) {
            return;
        }
        
        this.setState({isDisabled: this.props.isDisabled});
    }
    
    render() {
        ...
        ...
    }
}
```

Reply:
We have utilized React hooks in certain areas of our codebase, specifically in AvatarCropModal.js and PlaidLink/index.js. As the current component is a functional component, we are able to utilize React hooks in this instance. If, after consideration, we decide to convert OptionRow.js to a class component, I am able to undertake this task and begin the refactoring process.

Reply:
@syedsaroshfarrukhdot currently the only cases where we allow the use of hooks is where we are using a 3rd party library that provides no convenient alternative:

- In `AvatarCropModal.js`, we are using Reanimated's `useSharedValue` hook
- In `PlaidLink/index.js`, we are using react-plaid-links's `usePlaidLink` hook.

Don't get me wrong, I love hooks. It's just that, as a company, we haven't agreed to migrate our codebase over to using hooks yet (except in these very rare edge cases). In the meantime, we want to stick to class components to promote consistency across the codebase.

If/when we decide to migrate the codebase to hooks, we'll likely make a concerted effort to do it all as quickly as possible, because what we want to avoid is having a disjointed DX where sometimes we use hooks and sometimes we use class components, but it's not clear when to use which.

Reply:
So for now, please migrate your code to use a class component. Thanks!

Reply:
@roryabraham Sure will began the migration and will make the commit.

Reply:
Updated